### DISCUSSION TITLE
Appimage?

### DISCUSSION THREAD

**Discussion by robouden** [dune3d/dune3d#94] [Q&A]
Can we create an Appimage to linux?

Regards
Rob Oudendijk

**Comment by carrotIndustries** [comment]
From my point of view, the flatpak is sufficient.

If someone else wants to contribute an Appimage, sure go ahead.

**Reply by pallaswept** [reply]
For what it's worth, I can't try this because I don't have flatpak installed because they tend to require an insane amount of disk space for dependencies. That's a pretty common situation. An appimage would certainly be preferable.

**Comment by robouden** [comment]
Did not work for me.
Linux mint 21.3

Regards
Rob Oudendijk

**Comment by carrotIndustries** [comment]
> Did not work for me.

Please elaborate.

**Comment by robouden** [comment]
I installed by using flatpak, but if I run it nothing happens.. Seems some is missing?

Below are the erros:
`rob@rob-GS66-Stealth-10UG:~$ flatpak run org.dune3d.dune3d
realize
Compile failure in vertex shader: 0(2) : error C0201: unsupported version 330
0(3) : error C7532: global type vec2 requires "#version 100" or later
0(4) : error C7532: global type vec3 requires "#version 100" or later
0(4) : error C5060: out can't be used with non-varying color_to_fragment
0(9) : error C7532: global variable gl_VertexID requires "#version 130" or later
0(9) : error C0000: ... or #extension GL_EXT_gpu_shader4 : enable
0(10) : error C7532: global function pow requires "#version 100" or later
0(15) : error C7532: global variable gl_Position requires "#version 100" or later
0(15) : error C7532: global type vec4 requires "#version 100" or later
 
Compile failure in vertex shader: 0(2) : error C0201: unsupported version 330
0(3) : error C7532: global type vec3 requires "#version 100" or later
0(7) : error C5060: out can't be used with non-varying normal_to_fragment
0(8) : error C5060: out can't be used with non-varying color_to_fragment
0(9) : error C5060: out can't be used with non-varying pos_to_fragment
0(11) : error C7532: global type mat4 requires "#version 100" or later
0(14) : error C7532: global type mat3 requires "#version 100" or later
0(21) : error C7532: global function isnan requires "#version 130" or later
0(23) : error C7532: global type vec4 requires "#version 100" or later
0(26) : error C7532: global variable gl_Position requires "#version 100" or later
0(28) : error C7532: global function normalize requires "#version 100" or later
 
Compile failure in vertex shader: 0(2) : error C0201: unsupported version 330
0(3) : error C7532: global type vec3 requires "#version 100" or later
0(4) : error C7532: global type uint requires "#version 130" or later
0(4) : error C0000: ... or #extension GL_EXT_gpu_shader4 : enable
0(5) : error C7532: global type mat4 requires "#version 100" or later
0(7) : error C7532: global type float requires "#version 100" or later
0(9) : error C5060: out can't be used with non-varying pick_to_frag
0(10) : error C5060: out can't be used with non-varying color_to_frag
0(24) : error C0000: syntax error, unexpected '(', expecting "::" at token "("
0(32) : error C1503: undefined variable "colors"
0(32) : error C7548: '<<' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(32) : error C7548: '<<' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(32) : error C7548: '|' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(32) : error C7548: '<<' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(32) : error C7548: '|' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(32) : error C7548: '<<' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(32) : error C7548: '|' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(32) : error C7548: '<<' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(32) : error C7548: '|' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(32) : error C7548: '<<' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(32) : error C7548: '|' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(32) : error C7548: '&' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(46) : error C7548: '>>' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(46) : error C7548: '&' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(46) : error C7011: implicit cast from "uint" to "float"
0(47) : error C7548: '>>' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(47) : error C7548: '&' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(47) : error C7011: implicit cast from "uint" to "float"
0(48) : error C7548: '>>' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(48) : error C7548: '&' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(48) : error C7011: implicit cast from "uint" to "float"
0(49) : error C7548: '>>' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(49) : error C7548: '&' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(49) : error C7011: implicit cast from "uint" to "float"
0(57) : error C7532: global variable gl_Position requires "#version 100" or later
0(57) : error C7532: global type vec4 requires "#version 100" or later
0(58) : error C7532: global variable gl_VertexID requires "#version 130" or later
0(58) : error C0000: ... or #extension GL_EXT_gpu_shader4 : enable
0(58) : error C7532: global type int requires "#version 100" or later
 
Compile failure in vertex shader: 0(2) : error C0201: unsupported version 330
0(2) : error C0000: syntax error, unexpected '(', expecting "::" at token "("
0(6) : error C7532: global type vec4 requires "#version 100" or later
0(6) : error C5060: out can't be used with non-varying p1_to_geom
0(7) : error C5060: out can't be used with non-varying p2_to_geom
0(8) : error C7532: global type uint requires "#version 130" or later
0(8) : error C0000: ... or #extension GL_EXT_gpu_shader4 : enable
0(8) : error C5060: out can't be used with non-varying flags_to_geom
0(9) : error C5060: out can't be used with non-varying pick_to_geom
0(12) : error C7532: global type mat4 requires "#version 100" or later
0(14) : error C7532: global type float requires "#version 100" or later
0(28) : error C0000: syntax error, unexpected '(', expecting "::" at token "("
0(34) : error C7532: global type vec3 requires "#version 100" or later
0(36) : error C1503: undefined variable "colors"
0(36) : error C7548: '<<' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(36) : error C7548: '<<' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(36) : error C7548: '|' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(36) : error C7548: '<<' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(36) : error C7548: '|' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(36) : error C7548: '<<' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(36) : error C7548: '|' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(36) : error C7548: '<<' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(36) : error C7548: '|' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(36) : error C7548: '<<' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(36) : error C7548: '|' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(36) : error C7548: '&' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(50) : error C7548: '>>' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(50) : error C7548: '&' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(50) : error C7011: implicit cast from "uint" to "float"
0(51) : error C7548: '>>' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(51) : error C7548: '&' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(51) : error C7011: implicit cast from "uint" to "float"
0(52) : error C7548: '>>' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(52) : error C7548: '&' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(52) : error C7011: implicit cast from "uint" to "float"
0(53) : error C7548: '>>' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(53) : error C7548: '&' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(53) : error C7011: implicit cast from "uint" to "float"
0(59) : error C7532: global variable gl_VertexID requires "#version 130" or later
0(59) : error C0000: ... or #extension GL_EXT_gpu_shader4 : enable
0(59) : error C7532: global type int requires "#version 100" or later
0(60) : error C1503: undefined variable "flags"
0(61) : error C1503: undefined variable "flags"
0(61) : error C7548: '<<' requires "#extension GL_EXT_gpu_shader4 : enable" before use
0(62) : error C1503: undefined variable "p1"
0(64) : error C1503: undefined variable "p1"
0(67) : error C1503: undefined variable "p1"
0(70) : error C1503: undefined variable "p1"
0(73) : error C7532: global function length requires "#version 100" or later
0(73) : error C7532: global function max requires "#version 100" or later
0(74) : error C1503: undefined variable "p2"
0(74) : error C1503: undefined variable "p2"
0(74) : error C1503: undefined variable "p2"
0(79) : error C1503: undefined variable "p1"
0(80) : error C1503: undefined variable "p2"
 
Compile failure in vertex shader: 0(2) : error C0201: unsupported version 330
0(2) : error C0000: syntax error, unexpected '(', expecting "::" at token "("
0(8) : error C7532: global type vec4 requires "#version 100" or later
0(8) : error C5060: out can't be used with non-varying origin_to_geom
0(9) : error C7532: global type vec2 requires "#version 100" or later
0(9) : error C5060: out can't be used with non-varying shift_to_geom
0(10) : error C7532: global type float requires "#version 100" or later
0(10) : error C5060: out can't be used with non-varying scale_to_geom
0(11) : error C7532: global type uint requires "#version 130" or later
0(11) : error C0000: ... or #extension GL_EXT_gpu_shader4 : enable
0(11) : error C5060: out can't be used with non-varying flags_to_geom
0(12) : error C5060: out can't be used with non-varying pick_to_geom
0(13) : error C5060: out can't be used with non-varying bits_to_geom
0(16) : error C7532: global type mat4 requires "#version 100" or later
0(20) : error C7532: global variable gl_VertexID requires "#version 130" or later
0(20) : error C0000: ... or #extension GL_EXT_gpu_shader4 : enable
0(20) : error C7532: global type int requires "#version 100" or later
0(21) : error C1503: undefined variable "flags"
0(22) : error C1503: undefined variable "origin"
0(23) : error C1503: undefined variable "shift"
0(24) : error C1503: undefined variable "bits"
0(25) : error C1503: undefined variable "scale"
 
gl error 1,281 in ../src/canvas/glyph_renderer.cpp:44
`

**Reply by Jullyfish** [reply]
I have the same error. :(

I built dune3d on Fedora 38, it worked fine. Today I updated Fedora 38 -> Fedora 40 and got this error. I tried:
a) To rebuild the program.
b) sed -i 's/version 330/version 320 es/g' src/canvas/shaders/*
c) sed -i 's/version 330/version 150/g' src/canvas/shaders/*
d) rm -r ~/.cache/mesa_shader_cache/

But nothing helped. (After points b and c I got another error message but kind similar)

**Comment by robouden** [comment]
My experiences with Appmages is that they normally work. So simple to use. 

Regards
Rob Oudendijk

**Comment by priit123** [comment]
Flatpak gives the same error as above. Running KDE Neon 6.1

**Reply by Jullyfish** [reply]
Try this:
export GDK_DEBUG=gl-prefer-gl

**Reply by priit123** [reply]
Thanks. This worked perfectly.

**Comment by carrotIndustries** [comment]
See https://github.com/dune3d/dune3d/issues/87
